src/Core/Validation/FieldValidation/AbstractSymfonyFieldValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\ConstraintValidator;
+use SilverStripe\Core\Validation\FieldValidation\StringFieldValidator;
+
+/**
+ * Abstract class for validators that use Symfony constraints
+ */
+abstract class AbstractSymfonyFieldValidator extends StringFieldValidator
+{
+    protected function validateValue(): ValidationResult
+    {
+        $result = parent::validateValue();
+        if (!$result->isValid()) {
+            return $result;
+        }
+        $constraintClass = $this->getConstraintClass();
+        $args = [
+            ...$this->getContraintNamedArgs(),
+            'message' => $this->getMessage(),
+        ];
+        $constraint = new $constraintClass(...$args);
+        $validationResult = ConstraintValidator::validate($this->value, $constraint, $this->name);
+        return $result->combineAnd($validationResult);
+    }
+
+    /**
+     * The symfony constraint class to use
+     */
+    abstract protected function getConstraintClass(): string;
+
+    /**
+     * The named args to pass to the constraint
+     * Defined named args as assoc array keys
```

Comment:
```suggestion
     * Defined named args as associative array keys
```
Just to be a little kinder to anyone for whom English isn't their first language

Reply:
Has been removed